homeassistant/components/shelly/coordinator.py

```diff
@@ -460,21 +476,71 @@ def sw_version(self) -> str:
         """Firmware version of the device."""
         return self.device.firmware_version if self.device.initialized else ""
 
+    async def _async_disconnected(self) -> None:
+        """Handle device disconnected."""
+        async with self._connection_lock:
+            if not self.connected:  # Already disconnected
+                return
+            self.connected = False
+            self._async_run_disconnected_events()
+
     @callback
-    def _async_handle_update(self, device_: RpcDevice) -> None:
-        """Handle device update."""
-        device = self.device
-        if not device.initialized:
+    def _async_run_disconnected_events(self) -> None:
+        """Run disconnected events.
+
+        This will be executed on disconnect or when the config entry
+        is updated.
+        """
+        for disconnected_callback in self._disconnected_callbacks:
+            disconnected_callback()
+        self._disconnected_callbacks.clear()
+
+    async def _async_connected(self) -> None:
+        """Handle device connected."""
+        async with self._connection_lock:
+            if self.connected:  # Already connected
+                return
+            self.connected = True
+            await self._async_run_connected_events()
+
+    async def _async_run_connected_events(self) -> None:
+        """Run connected events.
+
+        This will be executed on connect or when the config entry
+        is updated.
+        """
+        await self._async_connect_ble_scanner()
+
+    async def _async_connect_ble_scanner(self) -> None:
+        """Connect BLE scanner."""
+        ble_scanner_mode = self.entry.options.get(
+            CONF_BLE_SCANNER_MODE, BLEScannerMode.DISABLED
+        )
+        if ble_scanner_mode == BLEScannerMode.DISABLED:
             return
-        event = device.event
-        status = device.status
+        if AwesomeVersion(self.device.version) < BLE_MIN_VERSION:
+            LOGGER.error(
```

Comment:
For a future PR we can make this a repair issue.

Reply:
Should probably do the same for esphome < 2022.11.x